view/js/dropzone-factory.js

```diff
@@ -0,0 +1,65 @@
+var DzFactory = function () {
+	this.createDropzone = function(dropSelector, textareaElementId) {
+		return new Dropzone(dropSelector, {
+			paramName: 'userfile', // The name that will be used to transfer the file
+			maxFilesize: max_imagesize, // MB
+			url: '/media/photo/upload?album=',
+			addRemoveLinks: true,
+			acceptedFiles: 'image/*',
+			clickable: true,
+			accept: function(file, done) {
+				done();
+			},
+			init: function() {
+				this.on('success', function(file, serverResponse) {
+					const targetTextarea = document.getElementById(textareaElementId);
+					const bbcodeString = serverResponse;
+					if (targetTextarea.setRangeText) {
+						//if setRangeText function is supported by current browser
+						targetTextarea.setRangeText(' ' + $.trim(bbcodeString) + ' ');
+					} else {
+						targetTextarea.focus();
+						document.execCommand('insertText', false /*no UI*/, '\n' + $.trim(bbcodeString) + '\n');
```

Comment:
```suggestion
					if (targetTextarea.setRangeText) {
						//if setRangeText function is supported by current browser
						targetTextarea.setRangeText(' ' + $.trim(serverResponse) + ' ');
					} else {
						targetTextarea.focus();
						document.execCommand('insertText', false /*no UI*/, '\n' + $.trim(serverResponse) + '\n');
```

Reply:
Changed it.

Reply:
Thanks, I'm still waiting on #12982 to be merged since this PR is now depending on it.